src/main/java/dev/codesoapbox/dummy4j/dummies/internet/PasswordBuilder.java

```diff
@@ -0,0 +1,153 @@
+        this.dummy4j = dummy4j;
+    }
+
+    /**
+     * Adds randomly generated digits
+     */
+    public PasswordBuilder withDigits() {
+        constraints.put("with digits", this::getDigit);
+        return this;
+    }
+
+    private Character getDigit() {
+        int number = dummy4j.number().nextInt(DIGIT_UPPER_BOUND);
+        return (char) (number + '0');
+    }
+
+    /**
+     * Adds randomly selected special characters
+     */
+    public PasswordBuilder withSpecialChars() {
+        constraints.put("with special characters", this::getSpecialChar);
+        return this;
+    }
+
+    private Character getSpecialChar() {
+        return dummy4j.expressionResolver().resolve(SPECIAL_CHAR_KEY).charAt(0);
+    }
+
+    /**
+     * Adds random upper case characters
+     */
+    public PasswordBuilder withUpperCaseChars() {
+        constraints.put("with upper case characters", this::getUpperCaseChar);
+        return this;
+    }
+
+    private Character getUpperCaseChar() {
+        return dummy4j.lorem().character().toUpperCase(Locale.ENGLISH).charAt(0);
+    }
+
+    /**
+     * Sets length of the generated password
+     */
+    public PasswordBuilder withLength(int length) {
+        this.minLength = length;
+        this.maxLength = length;
+        return this;
+    }
+
+    /**
+     * Sets minimum length of the generated password.
+     * If minimum length is greater than maximum length the password will be {@code minLength} long.
+     * Otherwise, the password will be between {@code minLength} and {@code maxLength} characters long.
+     */
+    public PasswordBuilder withMinLength(int minLength) {
+        if (minLength > maxLength) {
+            maxLength = minLength;
+        }
+        this.minLength = minLength;
+        return this;
+    }
+
+    /**
+     * Sets maximum length of the generated password.
+     * If maximum length is smaller than minimum length the password will be {@code maxLength} long.
+     * Otherwise, the password will be between {@code minLength} and {@code maxLength} characters long.
+     */
+    public PasswordBuilder withMaxLength(int maxLength) {
+        if (maxLength < minLength) {
+            minLength = maxLength;
+        }
+        this.maxLength = maxLength;
+        return this;
+    }
+
+    /**
+     * Returns a randomly generated password
+     */
+    public String build() {
+        checkMinLengthAgainstConstraints(constraints);
+        return buildPasswordWithConstraints(constraints);
+    }
+
+    private void checkMinLengthAgainstConstraints(Map<String, Supplier<Character>> constraints) {
+        if (constraints.size() >= minLength) {
+            String selectedConstraints = String.join(", ", constraints.keySet());
+            throw new IllegalArgumentException("Minimum length of the password must be GREATER than the number of " +
+                    "selected constraints. The following constraints were selected: [" + selectedConstraints + "] " +
```

Comment:
Add the number of selected constraints, e.g.
`Minimum length of the password must be GREATER than the number of selected constraints (2). [...]`